src/fromager/sources.py

```diff
@@ -311,17 +309,19 @@ def patch_source(
 
 
 def _warn_for_old_patch(
-    source_root_dir: pathlib.Path, patch_filename: pathlib.Path, version: Version
+    patches_dir: pathlib.Path, source_root_dir: pathlib.Path
 ) -> None:
-    # Get the existing version and req name as per the source_root_dir naming conventions
+    # Get the req name as per the source_root_dir naming conventions
     req_name = source_root_dir.name.rsplit("-", 1)[0]
-    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
 
-    # Check to see if patches for older version exists
-    if patch_filename.parent.name == source_root_dir.name:
-        logger.warning(
-            f"{req_name}: patches for version {existing_version!s} exist but will not be applied"
-        )
+    # Create expected path for patch file
+    patch_path = pathlib.Path(patches_dir / source_root_dir)
+
+    # Get all the patches for that requirement
+    req_patches = overrides.patches_for_req(patches_dir, req_name)
+    # Check if old patches exist and current patch not in the list
+    if len(req_patches) > 0 and patch_path not in req_patches:
+        logger.warning(f"{req_name}: old patches exist but will not be applied")
```

Comment:
Probably want to log the version of the patches not being applied (logging the patches directory might be enough)

Reply:
Done